### DISCUSSION TITLE
Informations about DrawNodeV2 (as draft: extension DrawNodeEx)

### DISCUSSION THREAD

**Discussion by aismann** [axmolengine/axmol#1949] [General]
My time to working on axmol/DrawNodeEx is slow at the moment.
`Extension DrawNodeEx` is mostly finished for replacing `Class DrawNode`.

1:  I think `DrawNodeEx `should be part of the cpp-test app (no longer optional) 
- "More developer who can see how its works and may finding an issue too".
For this: change this line to  ${AX_EXT_HINT}):
`option(AX_ENABLE_EXT_DRAWNODEEX "Build extension DrawNodeEx" ${AX_EXT_HINT}))`

2: After some testing time (4 weeks?) for everybody:
- rename `DraweNodeEx `to `Drawnode` and replace `Drawnode`. 
- Remove extension `DrawNodeEx`

**Comment by GamesStudioTwo** [comment]
quick questing on DrawNodeEx, is it a replacement  for DrawNode added to AXMOL.
DrawNode works perfect but can get a bit laggy like if I use Debug on box2d is DrawNodeEx faster or a replacement or has it always been there and I just never used it ?

**Reply by aismann** [reply]
> quick questing on DrawNodeEx, is it a replacement for DrawNode added to AXMOL. DrawNode works perfect but can get a bit laggy like if I use Debug on box2d is DrawNodeEx faster or a replacement or has it always been there and I just never used it ?

Best way is to try it self on cpp_test=>DrawNodeEx
DrawNode has some bugs and missing features.
Here a short list:
BUGs:
- #1888 DrawNode draws in wrong order
![image](https://github.com/axmolengine/axmol/assets/8652787/d201ac27-3bde-4316-86e3-ec3d7c53c75a)
- #1411 DrawNode: setLineWitdh() is no longer working 

Features (some was also started on DrawNode):
- All drawing primitives can be used on one node with different behavior (thickness, scale, rotating, position)
- drawPie() with thickness
- drawStar() 
- LineWith/thickness can be different for each drawing primitives
- Each drawing primitives can be transform (rotate, scale, position) without changing the thickness or recalc the vertices
- drawLine() can have different 'endtypes' 
![image](https://github.com/axmolengine/axmol/assets/8652787/77b7a3a2-92ca-42be-805b-e0517ffbda0f)

**Comment by aismann** [comment]
About the performance: It should be the same or little bit faster as DrawNode

**Comment by aismann** [comment]
DrawNodeV2 0.95.1:
![image](https://github.com/user-attachments/assets/b79ac2f2-b807-4191-8c14-478ad5167a97)
![image](https://github.com/user-attachments/assets/bfef93eb-8152-4ecf-84ae-0792267022d8)
![image](https://github.com/user-attachments/assets/4135be10-8797-45ff-aa00-d7aeff907740)
![image](https://github.com/user-attachments/assets/05331c60-1d30-4fb2-9aff-984b62b96ce7)
![image](https://github.com/user-attachments/assets/db34aebf-1618-4956-9ee5-a8b60488004f)
![image](https://github.com/user-attachments/assets/efdf8aa9-82de-41d9-bdf5-dab8a03fde83)

**Comment by aismann** [comment]
DrawNodeV2 0.96:
New example: Lines be generated with the same method `void Sample::generateDataPoints() `from: [AwesomeNode](https://github.com/intmainreturn00/AwesomeNode/)
![image](https://github.com/user-attachments/assets/6b4c6c62-c1c4-46fb-89f5-d194aa40a13d)

**Comment by aismann** [comment]
I closed it now
